pkg/middleware/request_metrics.go

```diff
@@ -59,8 +59,28 @@ func RequestMetrics(features featuremgmt.FeatureToggles, cfg *setting.Cfg, promR
 		histogramLabels,
 	)
 
+	nativeHTTPRequestDurationHistogram := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "grafana",
+			Name:      "native_http_duration_seconds",
+			Help:      "Native Histogram of latency for HTTP requests.",
+			// the recommended default value from the prom_client
+			// https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L411
+			// Giving this variable an value means the client will expose the histograms as an
+			// native histogram instead of normal a normal histogram.
+			NativeHistogramBucketFactor: 1.1,
```

Comment:
Not sure if you're aware: you can enable native histograms for an existing classic histogram in parallel, as the resulting series names will not clash. So potentially you could move these two lines to line 58 above, which would get you a native histogram on `http_request_duration_seconds` and the classic histogram under `http_request_duration_seconds_count` and `_sum`, `_bucket`s. Just make NativeHistogramBucketFactor depend on the feature flag, 0 if disabled and 1.1 if enabled.

Similar to how it was done for Mimir in dskit:
https://github.com/grafana/dskit/blob/d986a37b92cef35c5fb659f00567d8eae792bcab/server/metrics.go#L40

Also note the use of `NativeHistogramMinResetDuration: time.Hour,` there.

I'm writing the documentation on this, but got a bunch of comments from Beorn https://github.com/grafana/mimir/pull/5956

Reply:
Fixed!